lib/routes/wikipedia/current-events.ts

```diff
@@ -0,0 +1,260 @@
+            const depth = Math.floor(indent / 2) + 1;
+
+            // Close deeper levels
+            while (currentDepth > depth) {
+                result.push('  '.repeat(currentDepth - 1) + '</ul>');
+                openTags.pop();
+                currentDepth--;
+            }
+
+            // Open new level if needed
+            if (currentDepth < depth) {
+                result.push('  '.repeat(depth - 1) + '<ul>');
+                openTags.push('ul');
+                currentDepth = depth;
+            }
+
+            result.push('  '.repeat(depth) + trimmed);
+        } else {
+            // Close all open lists
+            while (currentDepth > 0) {
+                result.push('  '.repeat(currentDepth - 1) + '</ul>');
+                openTags.pop();
+                currentDepth--;
+            }
+
+            if (trimmed === '</p><p>') {
+                result.push('</p>\n<p>');
+            } else if (trimmed) {
+                result.push(trimmed);
+            }
+        }
+    }
+
+    // Close any remaining open lists
+    while (currentDepth > 0) {
+        result.push('  '.repeat(currentDepth - 1) + '</ul>');
+        currentDepth--;
+    }
+
+    return result.join('\n');
+}
+
+function wrapInParagraphsAndCleanup(html: string): string {
+    // Wrap in paragraphs and clean up
+    if (!html.startsWith('<p>')) {
+        html = '<p>' + html;
+    }
+    if (!html.endsWith('</p>')) {
+        html = html + '</p>';
+    }
+
+    // Clean up multiple paragraph tags and empty paragraphs
+    html = html.replaceAll(/<\/p>\s*<p>/g, '</p>\n<p>');
+    html = html.replaceAll(/<p>\s*<\/p>/g, '');
+    html = html.replaceAll(/<p>\s*<ul>/g, '<ul>');
+    html = html.replaceAll(/<\/ul>\s*<\/p>/g, '</ul>');
+
+    return html;
+}
+
+// Wiki markup to HTML converter with proper list handling
+function wikiToHtml(wikitext: string): string {
+    let html = wikitext;
+
+    // Apply transformations in order
+    html = convertWikiLinks(html);
+    html = convertExternalLinks(html);
+    html = convertTextFormatting(html);
+    html = processLines(html);
+    html = wrapListItems(html);
+    html = processNestedLists(html);
+    html = wrapInParagraphsAndCleanup(html);
+
+    return html;
+}
+
+function getCurrentEventsDatePath(date: Date): string {
+    const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+
+    const year = date.getFullYear();
+    const month = months[date.getMonth()];
+    const day = date.getDate(); // Not zero-padded
+
+    return `Portal:Current_events/${year}_${month}_${day}`;
+}
+
+async function fetchWikiRaw(pageName: string): Promise<string> {
+    const url = `https://en.wikipedia.org/wiki/${pageName}`;
+    const response = await got(url, {
+        searchParams: {
+            action: 'raw',
+        },
+        headers: {
+            'User-Agent': 'RSSHub/1.0 (+https://github.com/DIYgod/RSSHub)',
```

Comment:
Please use the provided value https://github.com/DIYgod/RSSHub/blob/380874aca8abfff9b1aad2cb77b50ecccbc77be5/lib/config.ts#L440 instead